Create initial consecutive http docs (EA) (#6637)
getsentry/sentry-docs: src/docs/product/issues/issue-details/performance-issues/consecutive-http/index.mdx

---
title: "Consecutive HTTP"
sidebar_order: 30
redirect_from:
  - /product/issues/performance-issues/consecutive-http/
description: "Learn more about Consecutive HTTP and how to diagnose and fix them."
---

<Include name="early-adopter-note.mdx" />

> Consecutive HTTP issues are created when a set of sequential http spans are consistently exceeding a duration threshold.

Reviewer: ```suggestion
Consecutive HTTP issues are created when a set of sequential HTTP spans consistently exceeds a duration threshold.
```

## Detection Criteria

> The detector for this performance issue looks for a set of sequential, non-overlapping http spans. It intentionally ignores non-http spans when finding this set, as the browser commonly performs other asynchronous operations such as loading resources, between http spans. 

Reviewer: ```suggestion
The detector for this performance issue looks for a set of sequential, non-overlapping HTTP spans. It intentionally ignores non-HTTP spans when finding this set, as the browser commonly performs other asynchronous operations such as loading resources, between HTTP spans. 
```

Once this sequence is found, the following must hold true for each HTTP span.

> - The http method must be a GET, POST, DELETE, PUT, PATCH
> - The http url must not begin with _next/static/ or _next/data/

Reviewer: ```suggestion
- The HTTP method must be a GET, POST, DELETE, PUT, PATCH
- The HTTP URL must not begin with _next/static/ or _next/data/
```

Once these spans are found, the following must also hold true:

> - The duration of each http span must exceed a threshold.
> - The time between each http span must exceed a threshold.
> - The number of sequential http spans must exceed a threshold.

Reviewer: ```suggestion
- The duration of each HTTP span must exceed a threshold.
- The time between each HTTP span must exceed a threshold.
- The number of sequential HTTP spans must exceed a threshold.
```

> If Sentry isn't detecting a Consecutive http issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.

Reviewer: ```suggestion
If Sentry doesn't detect a Consecutive HTTP issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.
```

## Span Evidence

> You can identify a Consecutive http problem by four main aspects in the "Span Evidence" section:

Reviewer: ```suggestion
You can identify a Consecutive HTTP problem by four main aspects in the "Span Evidence" section:
```

We say "four main aspects", but we only list two things. Is this a typo?

- Transaction name
- Sequential HTTP Spans

![Consecutive HTTP span evidence](span-evidence.png)

> View it by going to the **Issues** page in Sentry, selecting your project, clicking on the Consecutive HTTP Queries error you want to examine, then scrolling down to the "Span Evidence" section in the "Details" tab.

Reviewer: ```suggestion
You can view the span evidence by going to the "Issues" page in Sentry, selecting your project, selecting the Consecutive HTTP Queries error you want to examine, then scrolling down to the "Span Evidence" section in the "Details" tab.
```

## Fixing Consecutive HTTP Issues

> The reasons for a consecutive http issue varies, but here are some suggestions to help you resolve them.

Reviewer: ```suggestion
The causes of consecutive HTTP issues vary here are some suggestions to help you resolve them.
```

> - If each http span is independent of one another, consider parallelzing your http spans so they are no longer sequential.

Reviewer: ```suggestion
- If each HTTP span is independent from one another, consider parallelizing your HTTP spans so they are no longer sequential.
```

> - If you have control over the backend, consider to optimize your endpoints for this specific use case. You can dive into the corresponding backend transactions to help identify potential sources of improvement.

Reviewer: ```suggestion
- If you have control over the backend, consider optimizing your endpoints for this specific use case. You can investigate corresponding backend transactions to help identify potential sources of improvement.
```

> - The issue may also lie in the frontend as the span time does not always align with actual request time. This is because javascript won't always process an http request as soon as its response is ready due to it being asynchronous. There may be some other tasks that delay the time it takes for javascript to get to.

Reviewer: ```suggestion
- The issue may also lie in the frontend, as span time does not always align with actual request time. This is because JavaScript doesn't always process an HTTP request as soon as its response is ready because it's asynchronous and may be processing other tasks.
```

---
title: "Consecutive HTTP"
sidebar_order: 30
redirect_from:
  - /product/issues/performance-issues/consecutive-http/
description: "Learn more about Consecutive HTTP and how to diagnose and fix them."
---

<Include name="early-adopter-note.mdx" />

> Consecutive HTTP issues are created when a set of sequential HTTP spans consistently exceeds a duration threshold.

Reviewer 2: I've added the threshold now!

## Detection Criteria

The detector for this performance issue looks for a set of sequential, non-overlapping HTTP spans. It intentionally ignores non-HTTP spans when finding this set, as the browser commonly performs other asynchronous operations such as loading resources, between HTTP spans. 

Once this sequence is found, the following must hold true for each HTTP span:

- The HTTP method must be a GET, POST, DELETE, PUT, PATCH
- The HTTP URL must not begin with _next/static/ or _next/data/

Once these spans are found, the following must also hold true:

- The duration of each HTTP span must exceed a threshold.
- The time between each HTTP span must exceed a threshold.
- The number of sequential HTTP spans must exceed a threshold.

If Sentry doesn't detect a Consecutive HTTP issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.

## Span Evidence

> You can identify a Consecutive HTTP problem by four main aspects in the "Span Evidence" section:
> 
> - Transaction name
> - Sequential HTTP Spans
> 
> ![Consecutive HTTP span evidence](span-evidence.png)
> 
> You can view the span evidence by going to the "Issues" page in Sentry, selecting your project, selecting the Consecutive HTTP Queries error you want to examine, then scrolling down to the "Span Evidence" section in the "Details" tab.
> 

Reviewer 3: ```suggestion
You can find additional information about your Consecutive HTTP issue by looking at four main aspects in the "Span Evidence" section:

- **Transaction name**
- **Sequential HTTP Spans**

![Consecutive HTTP span evidence](span-evidence.png)

You can view the span evidence by going to the **Issues** page in Sentry, selecting your project, selecting the Consecutive HTTP Queries error you want to examine, then scrolling down to the "Span Evidence" section in the "Details" tab.

```
One big question - is saying "Sequential" and "Consecutive" the same thing? (Seeing it called "Sequential HTTP Spans in the image of the UI). Would be great if we were consistent.

Reviewer 2: Hmm, good question. I think we'd want to emphasize "consecutive", "sequential" is more about order but here we're more concerned about the fact that there are non-overlapping spans next to each other. I can make some changes to the wording here hopefully

Reviewer 2: Actually, I've been reading how we use "sequential" a bit more in these docs and I think it might make sense to keep these as is. There usually can be work that happens in-between the spans so they're technically not "next" to each other, but in sequence they are.

> ## Fixing Consecutive HTTP Issues
> 
> The causes of consecutive HTTP issues vary here are some suggestions to help you resolve them.
> - If each HTTP span is independent from one another, consider parallelizing your HTTP spans so they are no longer sequential.
> - If you have control over the backend, consider optimizing your endpoints for this specific use case. You can investigate corresponding backend transactions to help identify potential sources of improvement.
> - The issue may also lie in the frontend, as span time does not always align with actual request time. This is because JavaScript doesn't always process an HTTP request as soon as its response is ready because it's asynchronous and may be processing other tasks.

Reviewer 3: ```suggestion
## Fixing Consecutive HTTP Issues

The causes of Consecutive HTTP issues vary, but here are some suggestions to help you resolve them:

- If your HTTP spans are independent, consider parallelizing them so they're no longer sequential.
- If you have control over the backend, consider optimizing your endpoints for this specific use case. You can investigate corresponding backend transactions to help identify potential improvements.
- Since span time doesn't always align with actual request time, the issue may also be a frontend one. This is because JavaScript doesn't always process an HTTP request as soon as its response is ready because it's asynchronous and may be processing other tasks.
```

---
title: "Consecutive HTTP"
sidebar_order: 30
redirect_from:
  - /product/issues/performance-issues/consecutive-http/
description: "Learn more about Consecutive HTTP and how to diagnose and fix them."
---

<Include name="early-adopter-note.mdx" />

Consecutive HTTP issues are created when a set of sequential HTTP spans consistently exceeds 1000ms.

## Detection Criteria

<Note>

This issue is currently only available for backend and mobile SDKs.

</Note>

> The detector for this performance issue looks for a set of sequential, non-overlapping HTTP spans. It intentionally ignores non-HTTP spans when finding this set because the browser commonly performs other asynchronous operations, such as loading resources between HTTP spans.
> 

Reviewer 3: ```suggestion
The detector for this performance issue looks for a set of sequential, non-overlapping HTTP spans. It intentionally ignores non-HTTP spans because the browser commonly performs other asynchronous operations, such as loading resources between HTTP spans.

```

Once this sequence is found, the following must hold true for each HTTP span:

> - The HTTP method must be a GET, POST, DELETE, PUT, PATCH

Reviewer 3: ```suggestion
- The HTTP method must be a GET, POST, DELETE, PUT, or PATCH
```

- The HTTP URL must not begin with \_next/static/ or \_next/data/

Once these spans are found, the following must also hold true:

- The duration of each HTTP span must exceed a threshold of 1000ms.
- The time between each HTTP span must not exceed a threshold of 500ms.
- The number of sequential HTTP spans must exceed a threshold of 3.

If Sentry doesn't detect a Consecutive HTTP issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.

## Span Evidence

> You can find additional information about your Consecutive HTTP issue by looking at four main aspects in the "Span Evidence" section:

Reviewer 3: ```suggestion
You can find additional information about your Consecutive HTTP issue by looking at two main aspects in the "Span Evidence" section:
```
